File: web/siteplan/src/model/PositionedObject.ts
/**
 * Copyright (c) 2022 DB Netz AG and others.
 *
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v2.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v20.html
 */

import { checkInstance } from '@/util/ObjectExtension'
import SiteplanObject, { defaultObjectColorObj } from './SiteplanObject'
import { Position, defaultPositionObj } from './Position'

export default interface PositionedObject extends SiteplanObject {
    position: Position
}

export function defaultPositionedObj (): PositionedObject {
  return {
    guid: '0',
    position: defaultPositionObj(),
    objectColors: [defaultObjectColorObj()]
  }
}

// eslint-disable-next-line @typescript-eslint/no-explicit-any
export function isInstanceOfPositionedObject (object: any): boolean {
  return checkInstance(object, defaultPositionedObj())
}

File: java/bundles/org.eclipse.set.model.siteplan/src/org/eclipse/set/model/siteplan/PositionedObject.java
/**
 * Copyright (c) 2021 DB Netz AG and others.
 * 
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v2.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v20.html
 */
package org.eclipse.set.model.siteplan;

/**
 * <!-- begin-user-doc --> A representation of the model object
 * '<em><b>Positioned Object</b></em>'. <!-- end-user-doc -->
 *
 * <p>
 * The following features are supported:
 * </p>
 * <ul>
 * <li>{@link org.eclipse.set.model.siteplan.PositionedObject#getPosition
 * <em>Position</em>}</li>
 * </ul>
 *
 * @see org.eclipse.set.model.siteplan.SiteplanPackage#getPositionedObject()
 * @model
 * @generated
 */
public interface PositionedObject extends SiteplanObject {
	/**
	 * Returns the value of the '<em><b>Position</b></em>' containment
	 * reference. <!-- begin-user-doc --> <!-- end-user-doc -->
	 * 
	 * @return the value of the '<em>Position</em>' containment reference.
	 * @see #setPosition(Position)
	 * @see org.eclipse.set.model.siteplan.SiteplanPackage#getPositionedObject_Position()
	 * @model containment="true"
	 * @generated
	 */
	Position getPosition();

	/**
	 * Sets the value of the
	 * '{@link org.eclipse.set.model.siteplan.PositionedObject#getPosition
	 * <em>Position</em>}' containment reference. <!-- begin-user-doc --> <!--
	 * end-user-doc -->
	 * 
	 * @param value
	 *            the new value of the '<em>Position</em>' containment
	 *            reference.
	 * @see #getPosition()
	 * @generated
	 */
	void setPosition(Position value);

} // PositionedObject
